### PR TITLE
Initialization of the CePy Repo

### PR DESCRIPTION
**Pull request by tmarkovich**
This PR provides the initialization of the CePy repo. This includes the updating the License, adding the gitignore, adding requirements file, etc.

### CODE REVIEW

File: cepy/__init__.py
Comment (martinforsythe) [inline cepy/__init__.py]: Why are these 7 functions the only ones added to `__all__`?

File: cepy/cepy.py
Comment (martinforsythe) [inline cepy/cepy.py:124]: These functions want to have a short docstring that explains why they rename other functions. (I assume that it is to agree with scipy naming conventions?)

File: cepy/acosh.c
Comment (martinforsythe) [inline cepy/acosh.c:48]: May need/want to include documentation of our approval from Dr. Moshier to re-distribute Cephes

File: Makefile
Comment (martinforsythe) [inline Makefile]: I'd remove the sphinx and pytest benchmark stuff from the make file until/unless there are significant docs and benchmarks.

File: setup.py
Comment (martinforsythe) [inline setup.py]: `tests_require` should agree with the `requirements-test`

File: setup.py
Comment (martinforsythe) [inline setup.py]: Is this development status up to date?

File: setup.py
Comment (martinforsythe) [inline setup.py:43]: Has this been tested on all of these versions?

File: setup.py
Comment (martinforsythe) [inline setup.py]: looks like `codecs`, `os,` and `re` are not used and should be removed.

File: setup.py
Comment (martinforsythe) [reply setup.py]: From what I can see it looks like this wants to be:
```
tests_require = ['__future__', 'numpy', 'scipy']
```

File: tests/test_cephes.py
Comment (martinforsythe) [inline tests/test_cephes.py:31]: Why are you ignoring floating point errors for these tests?

File: cepy/cephes.h
Comment (martinforsythe) [inline cepy/cephes.h]: This header is out of date

File: requirements-test.txt
Comment (martinforsythe) [inline requirements-test.txt]: The tests also seem to have `numpy` as a requirement. I'd argue that `pytest` isn't actually a test requirement, just a requirement of the way that the Makefile has been setup.

File: Makefile
Comment (martinforsythe) [reply Makefile]: Actually, I'd remove all the pytest stuff and just run tests through setuptools.

**Review by martinforsythe — CHANGES_REQUESTED** [review]
They are mostly minor, but the repository cleanup comments should be addressed before this gets merged.

File: Makefile
Comment (jfinkels) [reply Makefile]: I'd remove the Makefile entirely :). Just do the minimum necessary work to get this new repository compile-able and test-able.

File: cepy/__init__.py
Comment (jfinkels) [inline cepy/__init__.py:2]: The Apache software foundation states that you should attach a specific header to each file that is part of the licensed software: https://www.apache.org/licenses/LICENSE-2.0#apply. In addition, it would be nice to have the first line of the header look like
```python
# nameoffile.py - one line describing purpose of file
```

File: cepy/acosh.c
Comment (jfinkels) [reply cepy/acosh.c:48]: Uh, I'm a little confused...is Cephes proprietary software? Do you have a written copyright license agreement to use and redistribute this software? How does that interact with the Apache license, specifically with the patents part?

File: cepy/build_cephes.py
Comment (jfinkels) [inline cepy/build_cephes.py:2]: Let's keep the source code for the Cephes library all in a single directory with no additional files. This way the Cephes distribution can just be plopped into a single directory if it needs to be updated, and it is clear from a copyright perspective that it is a single cohesive piece of software that has not been modified.

File: tests/test_cephes.py
Comment (jfinkels) [inline tests/test_cephes.py:34]: Have you considered using the built-in numpy.allclose function?

https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html

File: Makefile
Comment (martinforsythe) [reply Makefile]: On that note, is it worth adding information to the README about what this repository assumes in terms of your system's compiler?

File: tests/test_cephes.py
Comment (jfinkels) [inline tests/test_cephes.py]: Python doesn't want camel-case attributes.

File: tests/test_cephes.py
Comment (jfinkels) [inline tests/test_cephes.py:48]: There is a simpler way of having the same test with different arity functions: just have the test case specify the arity of the function in the `setUp()` (like `self.arity = 2`), and then do

```python
actual = starmap(self.fn, product(self.vals, repeat=self.arity))
expected = starmap(self.refFn, product(self.vals, repeat=self.arity))
```
Then you only need a single base mixin, like this:

```python
class ComparisonTestMixin(object):

    def assertAllClose(self, iterable1, iterable2):
        self.assertTrue(numpy.allclose(actual, expected))

    def test_fn(self):
        actual = starmap(self.fn, product(self.vals, repeat=self.arity))
        expected = starmap(self.refFn, product(self.vals, repeat=self.arity))
        self.assertAllClose(actual, expected)

class TestGamln(TestCase, ComparisonTestMixin):

    def setUp(self):
        self.arity = 1
        self.fn = gamln
        self.refFn = special.gammaln
        self.vals = [...]
```

File: cepy/__init__.py
Comment (tmarkovich) [reply cepy/__init__.py]: Added all of them

File: cepy/cepy.py
Comment (tmarkovich) [reply cepy/cepy.py:124]: done

File: tests/test_cephes.py
Comment (jfinkels) [inline tests/test_cephes.py]: This kind of test is basically an ad-hoc version of property-testing; see https://github.com/HypothesisWorks/hypothesis-python/

File: setup.py
Comment (jfinkels) [reply setup.py]: Don't need `__future__` there, it is part of Python. This metadata is for third-party dependencies only.

File: setup.py
Comment (jfinkels) [inline setup.py]: Why is this a TODO?

File: setup.py
Comment (jfinkels) [inline setup.py]: There's a bunch of copy and pasted code/comments that we don't need.

**Review by jfinkels — CHANGES_REQUESTED** [review]

File: cepy/build_cephes.py
Comment (martinforsythe) [reply cepy/build_cephes.py:2]: I agree. It will also make it clearer how to reference the licensing of Cephes.  A possible template for us could be the way that `torch-cephes` has done this https://github.com/deepmind/torch-cephes/blob/master/LICENSE.txt

File: cepy/__init__.py
Comment (martinforsythe) [reply cepy/__init__.py]: I'm not seeing the change?

File: cepy/build_cephes.py
Comment (jfinkels) [reply cepy/build_cephes.py:2]: Ah, that link helps with my comment above about Cephes being proprietary software.

File: cepy/__init__.py
Comment (tmarkovich) [reply cepy/__init__.py]: haven't pushed yet

File: Makefile
Comment (tmarkovich) [reply Makefile]: deleting right now

File: setup.py
Comment (tmarkovich) [reply setup.py]: Done

File: setup.py
Comment (tmarkovich) [reply setup.py]: Yes

File: setup.py
Comment (tmarkovich) [reply setup.py:43]: Yes

File: cepy/cephes.h
Comment (tmarkovich) [reply cepy/cephes.h]: done

File: requirements-test.txt
Comment (tmarkovich) [reply requirements-test.txt]: done

File: cepy/__init__.py
Comment (tmarkovich) [reply cepy/__init__.py:2]: Will add

File: setup.py
Comment (tmarkovich) [reply setup.py]: done

File: setup.py
Comment (tmarkovich) [reply setup.py]: done

File: tests/test_cephes.py
Comment (tmarkovich) [reply tests/test_cephes.py]: done

File: cepy/acosh.c
Comment (tmarkovich) [reply cepy/acosh.c:48]: Included the email

File: cepy/__init__.py
Comment (tmarkovich) [reply cepy/__init__.py:2]: Done

**Comment by tmarkovich** [comment]
@mbloodforsythe @jfinkels @mattbarr could I get a second look over this?

**Comment by jfinkels** [comment]
Sure, I'll take a look today.

File: LICENSE
Comment (jfinkels) [inline LICENSE:15]: The email is a little non-committal. What I was hoping for was "I agree to distribute Cephes under the terms of the Apache License, version 2.0." The email does not resolve the following issue: while we may be "including the material in your work by permission of the author", the author (i.e. the copyright holder) does not explicitly grant us the right to, among other things, redistribute the material. I'm sorry but I really think we need a lawyer to take a look at this one.

File: LICENSE
Comment (jfinkels) [inline LICENSE:3]: His email suggests including the original copyright notice, which I'm assuming means http://www.netlib.org/cephes/readme. Best to include that verbatim somewhere, perhaps nearby the email.

File: cepy/build_cephes.py
Comment (jfinkels) [reply cepy/build_cephes.py:2]: Thomas, the suggestion I made here for the directory structure still holds.

File: setup.py
Comment (jfinkels) [reply setup.py:43]: If this is true, let's add a `.travis.yml` to enable automatic testing for each of these Python versions. You could do that in a next pull request if you like.

**Review by jfinkels — COMMENTED** [review]
Some of my comments were not addressed, please double check some of the comments from the previous review.

**Comment by jfinkels** [comment]
Closing as I have some serious concerns about copyright license.

**Pull request closed without merging**
